File: src/sarma_cli/paths.py
"""Cross-platform path resolution for Sarma.

Two scopes:
- **global**  ``~/.sarma/``         — base config + skills, shared across projects.
                                        Override this directory with ``SARMA_HOME``
                                        (which then *is* the sarma dir, JAVA_HOME-style).
- **local**   ``./.sarma/``         — per-workspace overrides + session database.

``Path.home()`` resolves to ``C:\\Users\\<user>`` on Windows and ``/home/<user>``
(or ``/Users/<user>``) on Unix, so the same ``~/.sarma`` layout works everywhere.
"""

from __future__ import annotations

import os
from pathlib import Path

_DIR_NAME = ".sarma"
MODELS_NAME = "models.toml"
AGENTS_NAME = "agents.toml"
MCP_NAME = "mcp.toml"
INPUT_HISTORY_NAME = ".history"


def global_dir() -> Path:
    """Global config directory: ``$SARMA_HOME`` or ``~/.sarma``."""
    override = os.environ.get("SARMA_HOME")
    if override:
        return Path(override).expanduser()
    return Path.home() / _DIR_NAME


def local_dir() -> Path:
    """Per-workspace directory: ``./.sarma`` in the current working directory."""
    return Path.cwd() / _DIR_NAME


def global_config_file() -> Path:
    return global_models_file()


def local_config_file() -> Path:
    return local_models_file()


def global_models_file() -> Path:
    return global_dir() / MODELS_NAME


def local_models_file() -> Path:
    return local_dir() / MODELS_NAME


def global_agents_file() -> Path:
    return global_dir() / AGENTS_NAME


def local_agents_file() -> Path:
    return local_dir() / AGENTS_NAME


def global_mcp_file() -> Path:
    return global_dir() / MCP_NAME


def local_mcp_file() -> Path:
    return local_dir() / MCP_NAME


def db_path() -> Path:
    """Session database lives per-workspace."""
    return local_dir() / "db.sqlite"


def input_history_file() -> Path:
    """Line-based user input history for the full-screen TUI."""
    return local_dir() / INPUT_HISTORY_NAME


def global_skills_dir() -> Path:
    return global_dir() / "skills"


def local_skills_dir() -> Path:
    return local_dir() / "skills"

File: src/paths.ts
/**
 * Cross-platform path resolution for Sarma.
 *
 * Two scopes:
 * - **global**  `~/.sarma/`  — base config + skills, shared across projects.
 *                              Override with `SARMA_HOME` (which then *is* the
 *                              sarma dir, JAVA_HOME-style).
 * - **local**   `./.sarma/`  — per-workspace overrides + session database.
 */

import { homedir } from "node:os";
import { join } from "node:path";

const DIR_NAME = ".sarma";
export const MODELS_NAME = "models.toml";
export const AGENTS_NAME = "agents.toml";
export const MCP_NAME = "mcp.toml";
export const RAG_NAME = "rag.toml";
export const INPUT_HISTORY_NAME = ".history";

/** Global config directory: `$SARMA_HOME` or `~/.sarma`. */
export function globalDir(): string {
  const override = process.env.SARMA_HOME;
  if (override) return override;
  return join(homedir(), DIR_NAME);
}

/** Per-workspace directory: `./.sarma` in the current working directory. */
export function localDir(): string {
  return join(process.cwd(), DIR_NAME);
}

export function globalModelsFile(): string {
  return join(globalDir(), MODELS_NAME);
}
export function localModelsFile(): string {
  return join(localDir(), MODELS_NAME);
}
export function globalConfigFile(): string {
  return globalModelsFile();
}
export function localConfigFile(): string {
  return localModelsFile();
}
export function globalAgentsFile(): string {
  return join(globalDir(), AGENTS_NAME);
}
export function localAgentsFile(): string {
  return join(localDir(), AGENTS_NAME);
}
export function globalMcpFile(): string {
  return join(globalDir(), MCP_NAME);
}
export function localMcpFile(): string {
  return join(localDir(), MCP_NAME);
}
export function globalRagFile(): string {
  return join(globalDir(), RAG_NAME);
}
export function localRagFile(): string {
  return join(localDir(), RAG_NAME);
}
export function ragDir(): string {
  return join(localDir(), "rag");
}
export function globalRagDir(): string {
  return join(globalDir(), "rag");
}
export function ragDocsDir(): string {
  return join(ragDir(), "docs");
}
export function ragChromaDir(): string {
  return join(ragDir(), "chroma");
}
export function ragModelsDir(): string {
  return join(globalRagDir(), "models");
}
/** Session database lives per-workspace. */
export function dbPath(): string {
  return join(localDir(), "db.sqlite");
}
/** Line-based user input history for the full-screen TUI. */
export function inputHistoryFile(): string {
  return join(localDir(), INPUT_HISTORY_NAME);
}
export function globalSkillsDir(): string {
  return join(globalDir(), "skills");
}
export function localSkillsDir(): string {
  return join(localDir(), "skills");
}
